polyply/src/presistence.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2020 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import networkx as nx
+from .graph_utils import _compute_path_length_cartesian
+from .build_file_parser import apply_node_distance_restraints
+
+def worm_like_chain_model(h, L, _lambda):
+    """
+    Probability to find an end-to-end distance h given a contour length
+    L and presistence length _lambda according to the Worm-Like-Chain
+    model. The equation was published in: "Lee, H. and Pastor, R.W., 2011.
+    Coarse-grained model for PEGylated lipids: effect of PEGylation on the
+    size and shape of self-assembled structures. The Journal of Physical
+    Chemistry B, 115(24), pp.7830-7837."
+
+    Parameters
+    ----------
+    h: float
+        the end-to-end distance
+    L: float
+        the contour length
+    _lambda: float
+        the presistence length
+
+    Returns
+    -------
+    float
+        probability to find the end-to-end distance
+    """
+    alpha = 3*L/(4*_lambda)
+    C = (np.pi**3/2. * np.exp(-alpha)*alpha**(-3/2.)*(1 + 3/(alpha) + (15/4)/alpha**2.))**-1. 
+    A = 4*np.pi*h**2.*C
+    B = L*(1-(h/L)**2.)**9/2.
+    D = -3*L
+    E = 4*_lambda*(1-(h/L)**2.)
+    return A/B * np.exp(D/E)
+
+
+DISTRIBUTIONS = {"WCM": worm_like_chain_model, }
+
+def generate_end_end_distances(molecule, specs, nonbond_matrix, seed=None):
+    """
+    Subsample a distribution of end-to-end distances given a
+    presistence length, residue graph, and theoretical model.
+
+    Parameters
+    ----------
+    molecule: `class:nx.Graph`
+    specs: `tuple`
+        named tuple with attributes model, lp (i.e. presistence length),
```

Comment:
Not a tuple, and not just any namedtuple. You expect specific names/attributes, so list them